src/NuGet.Core/NuGet.Build.Tasks/RestoreTask.cs

```diff
@@ -192,5 +207,18 @@ private static MSBuildTaskItem GetMSBuildItem(ITaskItem item)
         {
             return new MSBuildTaskItem(item);
         }
+
+        private static void SetUserAgent()
+        {
+            var agent = "NuGet MSBuild Task";
+
+#if IS_CORECLR
+            UserAgent.SetUserAgentString(new UserAgentStringBuilder(agent)
+                .WithOSDescription(RuntimeInformation.OSDescription));
```

Comment:
[`RuntimeInformation`](https://docs.microsoft.com/en-us/dotnet/core/api/system.runtime.interopservices.runtimeinformation) is in the `System.Runtime.InteropServices` namespace, and yet I do not see a `using` directive for it.  I expect a compilation error here when `IS_CORECLR` is defined.